toolchain/check/impl_lookup.cpp

```diff
@@ -540,14 +540,15 @@ static auto CollectCandidateImplsForQuery(
     // TODO: We can skip the comparison here if the `impl_interface_const_id` is
     // not symbolic, since when the interface and specific ids match, and they
     // aren't symbolic, the structure will be identical.
-    if (!query_type_structure.CompareStructure(
+    if (!type_structure ||
```

Comment:
Can we split the `if (!type_structure) { continue }` into its own statement?

Reply:
Done